src/goto-instrument/contracts/dynamic-frames/dfcc_instrument.cpp

```diff
@@ -323,10 +323,20 @@ void dfcc_instrumentt::instrument_function_body(
 
   if(!goto_function.body_available())
   {
-    log.warning() << "DFCC instrumentation: '" << function_id
-                  << "' body is not available. Results may be unsound if the "
-                     "actual function has side effects."
-                  << messaget::eom;
+    // we interpret this as "the function should be unreachable"
+    // create fatal assertion code block as body
+    const auto &function_location =
+      utils.get_function_symbol(function_id).location;
+    source_locationt sl;
+    sl.set_property_class("reachability");
+    sl.set_function(function_id);
+    sl.set_line(0);
+    sl.set_column(0);
+    sl.set_file(function_location.get_file());
+    sl.set_working_directory(function_location.get_working_directory());
+    sl.set_comment(
+      "Function " + id2string(function_id) + " should not be reachable");
+    utils.gen_fatal_assertion_body(function_id, sl);
```

Comment:
Wouldn't it be possible to just directly use `generate_function_bodies`?

Reply:
I wanted to customise the error message and property class in that case.

Reply:
Is it really all that different? (I am really worried about the maintainability of this code.)

Reply:
now using `generate_function_bodies`